pinot-core/src/main/java/org/apache/pinot/core/operator/docidsets/AndDocIdSet.java

```diff
@@ -52,9 +54,12 @@
  */
 public final class AndDocIdSet implements FilterBlockDocIdSet {
   private final List<FilterBlockDocIdSet> _docIdSets;
+  private final boolean _cardinalityBasedRankingForScan;
 
-  public AndDocIdSet(List<FilterBlockDocIdSet> docIdSets) {
+  public AndDocIdSet(List<FilterBlockDocIdSet> docIdSets, Map<String, String> queryOptions) {
     _docIdSets = docIdSets;
+    _cardinalityBasedRankingForScan = queryOptions != null
+        && queryOptions.containsKey(OptimizationSwitches.AND_SCAN_CARDINALITY_BASED_REORDERING);
```

Comment:
Consider moving this to the `QueryOptionsUtils`. We usually parse the value for boolean flag

Reply:
Done